webapp/channels/src/components/edit_post/edit_post.tsx

```diff
@@ -348,7 +446,11 @@ const EditPost = ({editingPost, actions, canEditPost, config, channelId, draft,
         } else if (ctrlEnterKeyCombo) {
             handleEdit();
         } else if (Keyboard.isKeyPressed(e, KeyCodes.ESCAPE) && !showEmojiPicker) {
-            handleAutomatedRefocusAndExit();
+            if (scheduledPost) {
+                onCancel?.();
+            } else {
+                handleAutomatedRefocusAndExit();
+            }
```

Comment:
Why not call `handleCancel` here?

Reply:
Right,  there's no reason not to call it here. Done!